opm/material/localad/Evaluation.hpp

```diff
@@ -441,11 +443,27 @@ std::ostream& operator<<(std::ostream& os, const Evaluation<Scalar, VarSetTag, n
 // { std::cout << foo(0) << "\n"; }
 //
 // this will print '0' for bar() and '1' for foobar()...
-//
-#error "Due to some C++ peculiarity regarding function template specialization,"
-#error "the 'evaluation.hh' header must be included before Dune's 'densematrix.hh'!"
+#if !(DUNE_VERSION_NEWER(DUNE_COMMON, 2,4))
+
+#include "Math.hpp"
+
+namespace std {
+template <class Scalar, class VarSetTag, int numVars>
+const Opm::LocalAd::Evaluation<Scalar, VarSetTag, numVars> abs(const Opm::LocalAd::Evaluation<Scalar, VarSetTag, numVars>& x)
+{ return Opm::LocalAd::abs(x); }
+
+} // namespace std
+
+#if defined DUNE_DENSEMATRIX_HH
+#warning \
+ "Due to some C++ peculiarity regarding function overloads, the 'Evaluation.hpp'" \
+ "header file must be included before Dune's 'densematrix.hh' for Dune < 2.4. " \
+ "(If Evaluations are to be used in conjunction with a dense matrix.)"
+#endif
+
 #endif
 
+// this makes the Dune matrix/vector classes happy...
 #include <dune/common/ftraits.hh>
```

Comment:
Can we include this at the top instead of in the middle? I get nervous around such things...